algorithms/registration-toolbox/include/registration-toolbox/alignment/pcl-alignment.h

```diff
@@ -45,42 +55,137 @@ RegistrationResult PclAlignment<T_alignment, T_point>::registerCloudImpl(
   CHECK_NOTNULL(target);
   CHECK_NOTNULL(source);
 
-  PclPointCloudPtr<T_point> target_ds(new pcl::PointCloud<T_point>);
-  PclPointCloudPtr<T_point> source_ds(new pcl::PointCloud<T_point>);
+  PclPointCloudPtr<T_point> target_aligner_input;
+  PclPointCloudPtr<T_point> source_aligner_input;
 
-  voxel_grid_.setInputCloud(target);
-  voxel_grid_.filter(*target_ds);
-  voxel_grid_.setInputCloud(source);
-  voxel_grid_.filter(*source_ds);
+  if (FLAGS_regbox_pcl_use_downsampling) {
+    target_aligner_input =
+        PclPointCloudPtr<T_point>(new pcl::PointCloud<T_point>);
+    source_aligner_input =
+        PclPointCloudPtr<T_point>(new pcl::PointCloud<T_point>);
+    downSamplePointCloud(target, target_aligner_input);
+    downSamplePointCloud(source, source_aligner_input);
+  } else {
+    target_aligner_input = target;
+    source_aligner_input = source;
+  }
 
   bool is_converged = true;
   PclPointCloudPtr<T_point> transformed(new pcl::PointCloud<T_point>);
-  aligner_.setMaximumIterations(300);
-  aligner_.setRANSACIterations(1);
-  aligner_.setTransformationEpsilon(0.00001);
-  aligner_.setEuclideanFitnessEpsilon(1);
-  aligner_.setRANSACOutlierRejectionThreshold(0.05);
-  aligner_.setMaxCorrespondenceDistance(0.05);
-  aligner_.setUseReciprocalCorrespondences(false);
+  aligner_.setMaximumIterations(FLAGS_regbox_pcl_max_iterations);
 
   try {
     const Eigen::Matrix4f prior =
         prior_T_target_source.getTransformationMatrix().cast<float>();
-    aligner_.setInputTarget(target_ds);
-    aligner_.setInputSource(source_ds);
+    aligner_.setInputTarget(target_aligner_input);
+    aligner_.setInputSource(source_aligner_input);
     aligner_.align(*transformed, prior);
   } catch (std::exception& e) {
     LOG(ERROR) << "PCL registration failed to align the point clouds.";
     VLOG(3) << "PCL registration error: " << e.what();
     is_converged = false;  // just a procaution.
   }
+
+  const double fitness_score = aligner_.getFitnessScore(
```

Comment:
so far we only had the check for the fitness score in lpm matching, but I think it's good to have it so I added it for pcl as well